### DISCUSSION TITLE
Custom spi deployment in quarkus

### DISCUSSION THREAD

**Discussion by jfirles** [keycloak/keycloak#12541] [Archive]
I have custom provider for LoginForms, with name "myloginform". It's running in previous keycloak versions (16) but when I'm triying to migrate to the new version (18) I can't deploy it correctly.

I moved my jar to providers dir and I executed:
bin/kc.sh build --spi-login-forms-freemaker-enabled=false --spi-login-forms-myloginform-enabled=true --spi-login-forms-provider=myloginform

When I run with "start-dev", I can see how the "build" is executed another time (shows info for the jar detected) but when I'm trying to access, I get a NullPointerException because de LoginFormsProvider is null.

How I must deploy the jar? I follwed the doc steps with no luck.

**Comment by hongquyen196** [comment]
You can see my workaround for the custom provider deployed in Keycloak 18 Quarkus. #11973
